cmd/model_completion.go

```diff
@@ -49,3 +80,52 @@ func matchModel(prefix string) []string {
 	}
 	return nil // Nothing matched
 }
+
+// matchModel returns a list of tags with the given prefix.
+// If prefix is empty, it returns all tags.
+// A tag is in the format of key=value.
+func matchTag(prefix string) []string {
+	fileSys := getFileSystem()
+	if fileSys != nil {
+		tags := getAllTags(fileSys)
+		var matched []string
+		for tagName, files := range tags {
+			if strings.HasPrefix(tagName, prefix) {
+				// Include as suggestion:
+				//   tag:key=value -- Matches N models
+				matched = append(matched, fmt.Sprintf("tag:%s\tMatches %d models", tagName, len(files)))
+			}
+		}
+		// Sort output so consecutive autocompletes suggestions are stable
+		sort.Strings(matched)
+		return matched
+	}
+	return nil
+}
+
+// getAllTags reads all the tags on models.
+func getAllTags(fileSys *fs.FileSystem) map[string][]string {
```

Comment:
Different autocomplete invokes the binary separately, so it's not worth caching in memory, as  it'll be lost across completions. Maybe we could explore caching tags on file, but the parse/compile is fast enough to not need it right now.